src/main/java/org/spongepowered/mod/mixin/server/MixinServerConfigManager.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.mod.mixin.server;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.management.ServerConfigurationManager;
+import net.minecraft.util.ChatComponentTranslation;
+import net.minecraft.util.IChatComponent;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.gameevent.PlayerEvent;
+import org.spongepowered.api.event.entity.living.player.PlayerJoinEvent;
+import org.spongepowered.api.text.message.Message.Translatable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.mod.text.message.SpongeMessage;
+import org.spongepowered.mod.text.message.SpongeMessageTranslatable.SpongeMessageTranslatableBuilder;
+
+@Mixin(ServerConfigurationManager.class)
+public class MixinServerConfigManager {
+
+    private ChatComponentTranslation tmpJoinMessage;
+
+    // TODO Fix initializeConnectionToPlayer injection at runtime - it doesn't
+    // seem to know about FML's patches
```

Comment:
I'm not sure what we can do about this, this isn't actually the fault of your mixin or the AP. What's happening is that the patch is actually *altering the signature* of a public method, but the entry in the `srg` files is *not* updated with the corresponding information, so the `srg` still contains a mapping for the "old" (unpatched) version of the method.

This would be a problem for any code referencing that method (for example in a mod or mixin) because it means references won't get re-obfuscated and will thus be calling a non-existent method at production time. So basically the error is correct and we need to either:

1. Add the new reference the *srg* file so that it can be successfully reobfuscated, paging @AbrarSyed. A possible solution would be a companion file with the patches which specifies altered signatures which can be pumped into the srgs when they're generated.
2. Submit a PR to FML to fix this issue by doing whatever the patch is trying to to in a different way, since it could affect other code using that method as well.

Either way, it's not the mixin itself that needs to be fixed, and the AP error is telling us exactly what the issue is so the error is valid as well.